src/cluster_legacy.c

```diff
@@ -7029,48 +7029,73 @@ int clusterCommandSpecial(client *c) {
         clusterDelNode(n);
         clusterDoBeforeSleep(CLUSTER_TODO_UPDATE_STATE | CLUSTER_TODO_SAVE_CONFIG);
         addReply(c, shared.ok);
-    } else if (!strcasecmp(c->argv[1]->ptr, "replicate") && c->argc == 3) {
+    } else if (!strcasecmp(c->argv[1]->ptr, "replicate") && (c->argc == 3 || c->argc == 4)) {
         /* CLUSTER REPLICATE <NODE ID> */
         /* Lookup the specified node in our table. */
-        clusterNode *n = clusterLookupNode(c->argv[2]->ptr, sdslen(c->argv[2]->ptr));
-        if (!n) {
-            addReplyErrorFormat(c, "Unknown node %s", (char *)c->argv[2]->ptr);
-            return 1;
+        clusterNode *n = NULL;
+        if (c->argc == 4) {
```

Comment:
can you try to do a fast path in here? the diff is kind of ugly and hard to read.

Reply:
I'm sorry for naive question, but what is a fast path? Is this some kind of formatting tool?

Reply:
like simply just move this block to here and do a return, so you dont have the change the other lines, so the diff will be smaller
```
if (n == NULL) {
            if (nodeIsPrimary(myself)) {
                addReply(c,shared.ok);
                return 1;
            }
            serverLog(LL_NOTICE,"Stop replication and turning myself into empty primary.");
            clusterSetNodeAsPrimary(myself);
            if (server.primary != NULL) {
                replicationUnsetPrimary();
            }
....
}
```

Reply:
Updated.